### PR TITLE
Fix device limitations in `podman-remote update` on remote systems

### PR DESCRIPTION
**Pull request by Honny1**
This PR fixes device limitations using `podman-remote update` on remote systems. The cause of the problem was that the major and minor numbers were loaded on the client system and then sent to the remote system, where there could be different major and minor numbers for each device. 

Reproducer in Issue: #24734

Fixes: https://issues.redhat.com/browse/RUN-2381
Fixes: https://github.com/containers/podman/issues/24734

#### Does this PR introduce a user-facing change?

```release-note
Fix device limitations in `podman-remote update` on remote systems
```

### CODE REVIEW

File: pkg/domain/entities/types/containers.go
Comment (Luap99) [inline pkg/domain/entities/types/containers.go:44]: This type is used in pkg/bindings/containers Update() function as such we cannot chnage the type as this will break bindings users, we can add new fields but we cannot remove the function of old fields.

**Review by Luap99 — CHANGES_REQUESTED** [review]

File: pkg/domain/entities/types/containers.go
Comment (Luap99) [inline pkg/domain/entities/types/containers.go:51]: @mheon PTAL I never liked using specgen here as it is totally unclear what field are or are not supported so I am fine with this but I am not sure if have objects?

File: pkg/domain/entities/types/containers.go
Comment (Luap99) [inline pkg/domain/entities/types/containers.go:50]: Any reason to define this `*string`? It seems pretty clear to me that `string` should work fine too. Just leave the string empty if it should not be changed sounds easy enough. I don't see what the extra pointer offers here? Allowing user to set the restart policy to an empty string seems pointless as this is not valid?

File: pkg/domain/entities/types/containers.go
Comment (mheon) [reply pkg/domain/entities/types/containers.go:51]: We're not actually using the full thing, so passing a full Specgen feels deceptive. A lot easier to say what we actually support like this.

File: pkg/domain/entities/types/containers.go
Comment (Honny1) [reply pkg/domain/entities/types/containers.go:50]: I'm sure an empty string would work, but the intent was to be consistent in the representation of the unset value throughout the structure.

File: pkg/domain/entities/types/containers.go
Comment (Honny1) [reply pkg/domain/entities/types/containers.go]: Updated the doc string with the list of fields that are currently used from `SpecGenerator` struct.

**Comment by mheon** [comment]
Changes LGTM

**Comment by rhatdan** [comment]
@Honny1 @Luap99 what is the state of this PR?

**Comment by Honny1** [comment]
@rhatdan PR is waiting for a review from @Luap99. @Luap99 is on PTO.

**Pull request merged**
